lib/use.js

```diff
@@ -0,0 +1,24 @@
+const {
+  all, compose, curryN, ifElse, nAry, once, partial, props, uncurryN
+} = require('ramda')
+
+const { tapP } = require('@articulate/funky')
+
+// applyMiddleware :: ((IncomingMessage, HttpResponse, Function) -> ()) -> Request -> Request
+const applyMiddleware = uncurryN(2, middle => tapP(({ request, response }) =>
+  new Promise(partial(middle, [ request, response ]))
+))
+
+// warn :: String -> * -> ()
+const warn =
+  curryN(2, once(nAry(1, console.warn)))
```

Comment:
Just double-checking, because I think I've deciphered it properly, but this only logs the warning string, not the whole request, right?

Reply:
Correct.